services/src/main/scala/org/apache/gearpump/services/AppMasterService.scala

```diff
@@ -51,13 +54,7 @@ trait AppMasterService extends HttpService  {
            case true =>
              onComplete((master ? AppMasterDataDetailRequest(appId)).asInstanceOf[Future[AppMasterDataDetail]]) {
                case Success(value: AppMasterDataDetail) =>
-                 if (isStreamingApplication(value)) {
-                   val streamApp: StreamingAppMasterDataDetail = value
-                   complete(write(streamApp))
-                 } else {
-                    val app: GeneralAppMasterDataDetail = value
-                   complete(write(app))
-                 }
+                 complete(value.toJson)
```

Comment:
Good idea!

I would suggest to wrap the json library in a Util in the future .
Like 

object JsonUtil {
  def toJson(obj: T)
  def fromJson(json: String): T
}